rust/src/ffiutil.rs

```diff
@@ -37,7 +37,7 @@ use openat;
 /// Convert a C (UTF-8) string to a &str; will panic
 /// if it isn't valid UTF-8.  Note the lifetime of
 /// the return value must be <= the pointer.
-pub fn str_from_nullable<'a>(s: *const libc::c_char) -> Option<&'a str> {
+pub fn ffi_view_nullable_str<'a>(s: *const libc::c_char) -> Option<&'a str> {
     if s.is_null() {
         None
     } else {
```

Comment:
Maybe let's also make it convention to use `expect()` instead of `unwrap()` for these new functions like we do in `ffi_view_openat_dir`? So e.g. here:

```rust
Some(s.to_str().expect("ffi_view_nullable_str: valid utf-8"))
```

OTOH, if it's a macro through `assert!()` then that's fine since the expression itself will show up in the panic msg.